src/Containers/StoreSelector/StoreSelector.tsx

```diff
@@ -0,0 +1,205 @@
+    /* Name of the shop */
+    storeSelectorName: string,
+    /* Id of the store, servers as array key */
+    storeSelectorId: number,
+    /* Status of the store, border changes colour depnding on sates */
+    storeStatus: VendorApprovalStatus,
+}
+
+export interface StoreSelectorProps
+    extends React.HTMLAttributes<HTMLDivElement> {
+    /* Store name, store ID, and store status used from storeSelectorProps */
+    storeProfileData: StoreSelectorArrayProps[];
+    /* For use of VendorApprovalStatus from current object */
+    storeBorderStatus?: VendorApprovalStatus,
+    /* Toggles the up arrow and up arrow click action */
+    upArrowEnabled?: boolean,
+    /* Toggles the down arrow and down click action */
+    downArrowEnabled?: boolean,
+    /* General onclick handler */
+    onClickArrows?: () => void,
+    /* Up arrow click handler */
+    onClickUp?: () => void,
+    /* Down arrow click handler */
+    onClickDown?: () => void, 
+}
+
+export const StoreSelector: React.FC<StoreSelectorProps> = ({
+    storeProfileData,
+    onClickArrows,
+    onClickUp,
+    onClickDown,
+    downArrowEnabled,
+    upArrowEnabled, 
+    ...props
+}): React.ReactElement => {
+    const [currentIndex, setCurrentIndex] = useState(0);
+    /* Gets the props for the current store by comparing its storeSelectorId to the current index state */
+    const currentCardObject = storeProfileData.find(storeProfileData => storeProfileData.storeSelectorId === currentIndex);
+    /**
+    * Takes the state, and increases the value by 1 conditionally. 
+    * If the value is equal to the size of the array, then the value gets set to the bottom of the array instead (loops back around)
+    * @param currentIndex - Stateful value, used to get the current object of the array
+    * @param setCurrentIndex - Function to update currentIndex stateful value
+    * @param storeProfileData - Array that currentIndex is used on
+    */
+    let indexIncrease = (currentIndex: number, setCurrentIndex: Function, storeProfileData: Array<StoreSelectorArrayProps>) => {(
+            currentIndex === (storeProfileData.length - 1)
+                ? setCurrentIndex(currentIndex - (storeProfileData.length - 1))
+                : setCurrentIndex(currentIndex + 1)
+    )}
+    /**
+     * Takes the state, and decreases the value by 1 conditionally.
+     * If the value is equal to 0, then value gets set to the top of the array instead (loops back around)
+     * @param currentIndex - Stateful value, used to get the current object of the array
+     * @param setCurrentIndex - Function to update that stateful value
+     * @param storeProfileData - Array that currentIndex is used on
+     */
+    let indexDecrease = (currentIndex: number, setCurrentIndex: Function, storeProfileData: Array<StoreSelectorArrayProps>) => {(
+            currentIndex === 0
+                ? setCurrentIndex(currentIndex + (storeProfileData.length - 1))
+                : setCurrentIndex(currentIndex - 1)
```

Comment:
The setState should only be called once, use a variable to change the value and call setState at the end